File: Math/Medium/50/50.js
/**
 * @param {number} x
 * @param {number} n
 * @return {number}
 */
var myPow = function(x, n) {
  if (n === 0) {
    return 1;
  }

  if (n < 0) {
    x = 1 / x;
    n = -n;
  }

  if (n % 2 === 0) {
    let halfPower = myPow(x, n / 2);
    return halfPower * halfPower;
  }
  else {
    return x * myPow(x, n - 1);
  }  
}

File: Math/Medium/50/50.c
double myPow(double x, int n) {
  if(n == 0){
    return 1;
  }

  if(n < 0){
    x = 1 / x;
    n = -n;
  }

  if(n % 2 == 0){
    double halfPower = myPow(x, n / 2);
    return halfPower * halfPower;
  }else{
    return x * myPow(x, n - 1);
  }
}